src/shogun/base/SGObject.h

```diff
@@ -817,13 +817,28 @@ class CSGObject
 		BaseTag tag(name);
 		AnyParameterProperties properties(
 			"Dynamic parameter",
-			ParameterProperties::HYPER |
-			ParameterProperties::GRADIENT |
-            ParameterProperties::MODEL);
+			ParameterProperties::NONE);
 		std::function<T()> bind_method =
 			std::bind(method, dynamic_cast<const S*>(this));
 		create_parameter(tag, AnyParameter(make_any(bind_method), properties));
 	}
+
+	/** Puts a pointer to a (lazily evaluated) function into the parameter map.
+	 *
+	 * @param name name of the parameter
+	 * @param method pointer to the method
+	 */
+	template <typename T, typename S>
+	void watch_method(const std::string& name, T (S::*method)())
```

Comment:
@lisitsyn is it OK to have a watch_method for non const functions?

Reply:
Ah I didn’t think about the function being non const of course. Another reason to have a special directive (run) as it might change the state of the object.

Reply:
@karlnapf it seems to be straight forward, just have to decide how two void functions compare, i.e. if `f1` and `f2` are void functions is `f1() == f2()`?